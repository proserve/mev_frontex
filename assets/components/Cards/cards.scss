.container {
  margin: 30px 20px;
}

.breadcrumbs {
  font-size: 1.1em;
  text-transform: uppercase;
  font-weight: 500;
  word-spacing: 1.5px;
  letter-spacing: 1px;
  margin-right: 30px;

  .left {
    color: #89898a;
  }
}

.circle {
  width: 16px;
  height: 16px;
  border-radius: 30px;
  margin: 5px 20px 5px 5px;
}

.details {
  margin-top: 20px;
  margin-right: 30px;

  .title {
    font-size: 1.25em;
    font-weight: 300;
    text-transform: uppercase;
    letter-spacing: 1px;
    padding-left: 20px;
  }

  .boxes {
    display: flex;
    flex-wrap: wrap;
    margin: 10px -20px 0px;
    .box {
      display: flex;
      align-items: center;
      background-color: white;
      float: left;
      padding: 15px;
      border-radius: 6px;
      position: relative;
      box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
      transition: all 0.3s cubic-bezier(.25, .8, .25, 1);
      margin-bottom: 10px;
      margin-left: 20px;
      width: 280px;
      &:hover {
        box-shadow: 0 5px 10px rgba(0, 0, 0, 0.25), 0 5px 5px rgba(0, 0, 0, 0.22);
        cursor: pointer;
      }

      .yellow-circle {
        @extend .circle;
        background-color: #f7c057;
      }

      .green-circle {
        @extend .circle;
        background-color: #009a3c;
      }

      .name {
        margin-bottom: 5px;
        font-size: 1.2em;
      }

      .price {
        font-size: 1.8em;
        font-weight: 500;
      }
    }
  }
}

