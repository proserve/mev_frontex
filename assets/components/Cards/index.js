import React from 'react';
import PropTypes from 'prop-types';

import './cards.scss';

export default class Cards extends React.Component {
  render() {
    const { cards } = this.props;
    return (
      <div className="boxes">
        {cards.map(({ color, name, price }) => (
          <div className="box" key={name + price}>
            <div className={`${color}-circle`} />
            <div>
              <div className="name">{name}</div>
              <div className="price">{price}</div>
            </div>
          </div>
        ))}
      </div>
    );
  }
}


Cards.propTypes = {
  cards: PropTypes.arrayOf(PropTypes.object),
};

Cards.defaultProps = {
  cards: [],
};
