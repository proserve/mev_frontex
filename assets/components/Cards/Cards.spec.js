import React from 'react';
import 'isomorphic-fetch';
import { shallow } from 'enzyme';
import Cards from './index';

describe('Cards', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Cards />);
    expect(component).toMatchSnapshot();
  });
});
