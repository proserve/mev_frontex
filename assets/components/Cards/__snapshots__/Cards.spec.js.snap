// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should render correctly in "debug" mode 1`] = `
<div
  className="boxes"
/>
`;
